controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller.go

```diff
@@ -0,0 +1,185 @@
+	// in that case we can skip it
+	if hasSpaceBindingRequest(spaceBinding) {
+		return reconcile.Result{}, nil
+	}
+
+	spaceName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.Space}
+	space := &toolchainv1alpha1.Space{}
+	if err := r.Client.Get(ctx, spaceName, space); err != nil {
+		if errors.IsNotFound(err) {
+			// space was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading space
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound Space")
+	}
+
+	murName := types.NamespacedName{Namespace: spaceBinding.Namespace, Name: spaceBinding.Spec.MasterUserRecord}
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	if err := r.Client.Get(ctx, murName, mur); err != nil {
+		if errors.IsNotFound(err) {
+			// mur was deleted
+			return reconcile.Result{}, nil
+		}
+		// error while reading MUR
+		return ctrl.Result{}, errs.Wrapf(err, "unable to get the bound MUR")
+	}
+
+	// skip workspace creator spacebinding
+	// the controller will convert only spacebindings created by system admins using the sandbox-cli.
+	// If the creator label on the space matches the usersignup reference on the MUR then this is the owner of the space
+	// and the spacebinding should not be migrated.
+	if mur.ObjectMeta.OwnerReferences == nil || len(mur.ObjectMeta.OwnerReferences) == 0 {
+		return ctrl.Result{}, errs.New("MasterUserRecord has no UserSignup owner reference")
+	}
+	// skip spaces with no creator label
+	if _, ok := space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey]; !ok {
+		// let's log an error, since this should not happen in production
+		logger.Error(errs.New("space has no SpaceCreatorLabelKey set"), "the spacebindings for this space will not be migrated", "space name", space.Name)
+		return ctrl.Result{}, nil
+	}
+	usersignup := mur.ObjectMeta.OwnerReferences[0]
+	if space.Labels[toolchainv1alpha1.SpaceCreatorLabelKey] == usersignup.Name {
+		return reconcile.Result{}, nil
+	}
+
+	// get the spaceRole
+	spaceRole := spaceBinding.Spec.SpaceRole
+
+	// get member cluster name where the space was provisioned
+	targetCluster := space.Spec.TargetCluster
+	memberCluster, memberClusterFound := r.MemberClusters[targetCluster]
+	if !memberClusterFound {
+		return ctrl.Result{}, errs.New(fmt.Sprintf("unable to find member cluster: %s", targetCluster))
+	}
+
+	// get the home namespace from space
+	defaultNamespace := getDefaultNamespace(space.Status.ProvisionedNamespaces)
+
+	// construct a SpaceBindingRequest object
+	sbrName := mur.GetName() + "-" + spaceRole
+	sbr := &toolchainv1alpha1.SpaceBindingRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sbrName,
+			Namespace: defaultNamespace,
+			Labels: map[string]string{
+				toolchainv1alpha1.SpaceCreatorLabelKey:                 "spacebindingrequestmigration",
+				toolchainv1alpha1.SpaceBindingSpaceLabelKey:            space.GetName(),
+				toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey: mur.GetName(),
```

Comment:
We don't use these labels in case of SBRs anywhere either, right? I would drop them too, to not confuse anyone.

Reply:
Not used right! Removed , please see https://github.com/codeready-toolchain/host-operator/pull/922/commits/532ca3aed1ef475bb9381fd31e1287715d2d73bb